otherlibs/unix/unixLabels.mli

```diff
@@ -1386,24 +1390,24 @@ type group_entry = Unix.group_entry =
   }
 (** Structure of entries in the [groups] database. *)
 
-val getlogin : unit -> string
+val getlogin : unit -> string @@ nonportable
 (** Return the login name of the user executing the process. *)
 
-val getpwnam : string -> passwd_entry
+val getpwnam : string -> passwd_entry @@ nonportable
 (** Find an entry in [passwd] with the given name.
    @raise Not_found if no such entry exists, or always on Windows. *)
 
-val getgrnam : string -> group_entry
+val getgrnam : string -> group_entry @@ nonportable
 (** Find an entry in [group] with the given name.
 
    @raise Not_found if no such entry exists, or always on Windows. *)
 
-val getpwuid : int -> passwd_entry
+val getpwuid : int -> passwd_entry @@ nonportable
 (** Find an entry in [passwd] with the given user id.
 
    @raise Not_found if no such entry exists, or always on Windows. *)
 
-val getgrgid : int -> group_entry
+val getgrgid : int -> group_entry @@ nonportable
 (** Find an entry in [group] with the given group id.
 
    @raise Not_found if no such entry exists, or always on Windows. *)
```

Comment:
`gethostbyname`, `gethostbyaddr` is also kinda unsafe, if the `_r` versions exist they're fine but not otherwise.
`getprotobyname`, `getprotobynumber`, `getservbyname`, `getservbyport` are all racy.

Reply:
I marked `getprotobyname`, `getprotobynumber`, `getservbyname`, and `getservbyport` as nonportable.

I believe we can assume the `_r` versions exists.

To make `getaddrinfo` and `getnameinfo` portable, I made it so that, in case the safe versions don't exist, the "emulation" using thread-unsafe functions is allowed only when running on main domain.